code/game/machinery/kitchen/smartfridge.dm

```diff
@@ -294,7 +294,7 @@
 			thisPile.addAmount(1)
 		else
 			piles[O.name] = new /datum/fridge_pile(O.name, src, 1, costly_bicon(O))
-		if(user)
+		if(!quiet)
```

Comment:
It needs to still check `user` too. Otherwise any automated insertions (such as fridges that start loaded) will runtime.

Reply:
No, it returns at line 289 if there is no user, that case is not possible

Reply:
Only if they also fail to drop the item.

Reply:
God damn it I looked there but was tired and thought it was backwards

Reply:
You have me second guessing it, it SHOULD return but how does the insertion get handled otherwise?

I'm just gonna test it locally